### DISCUSSION TITLE
trouble shooting

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#735] [Q&A] [answered]
In your example i2s to seral .  I replace coper.copy with below 4 line for some trouble shooing .

 file is recorded but its audio is very strange and disturb .  seem there is some sampling issue. but  don't now why 
below is audacity mage and recorded file
......
int32_t raw_samples[SAMPLE_BUFFER_SIZE]; 
size_t bytes_read = i2sStream.readBytes((uint8_t *)raw_samples, sizeof(int32_t) * SAMPLE_BUFFER_SIZE);
int samples_read = bytes_read / sizeof(int32_t);
dataFile.write( (uint8_t *)raw_samples , samples_read ) ;
.............
18:45:12.211 -> [I] I2SConfig.h : 106 - rx/tx mode: RX_MODE
18:45:12.211 -> [I] I2SConfig.h : 107 - port_no: 0
18:45:12.211 -> [I] I2SConfig.h : 108 - is_master: Master
18:45:12.211 -> [I] I2SConfig.h : 109 - sample rate: 44100
18:45:12.211 -> [I] I2SConfig.h : 110 - bits per sample: 32
18:45:12.258 -> [I] I2SConfig.h : 111 - number of channels: 1
18:45:12.258 -> [I] I2SConfig.h : 112 - i2s_format: I2S_STD_FORMAT
18:45:12.258 -> [I] I2SConfig.h : 114 - auto_clear: false
18:45:12.258 -> [I] I2SConfig.h : 121 - buffer_count:6
18:45:12.258 -> [I] I2SConfig.h : 122 - buffer_size:512
18:45:12.258 -> [I] I2SConfig.h : 126 - pin_bck: 14
18:45:12.258 -> [I] I2SConfig.h : 128 - pin_ws: 15
18:45:12.258 -> [I] I2SConfig.h : 130 - pin_data: 32

![image](https://user-images.githubusercontent.com/65456998/229135802-a00894aa-9bde-4006-8c6d-ad38e14449ef.png)

[example.txt](https://github.com/pschatzmann/arduino-audio-tools/files/11122375/example.txt)

**Comment by aqildad-create** [comment]
above is 15 second recording  

so file size should be  = 44100 (sampling frequency )  x 4 ( sampling bit 32 )  x 15 (seconds) = 2646000 bytes 
while file size i get is 658,432 bytes .

with copier.copy  I get the exact size , 

so what make the size reduce ? their is not too much processing in the loop  just read and write .

**Comment by pschatzmann** [comment, marked as answer]
Files work with bytes and not with samples! Your size is completley wrong:
it nees to be dataFile.write( (uint8_t *)raw_samples , bytes_read ) ;

You should find these kind of errors yourself!
